buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy

```diff
@@ -265,23 +263,46 @@ class DistributedPerformanceTest extends PerformanceTest {
         return null
     }
 
+    void waitForTestsCompletion() {
+        int total = scheduledBuilds.size()
+        Set<String> completed = []
+        while (completed.size() < total) {
+            scheduledBuilds.each { build ->
+                if (!completed.contains(build)) {
+                    if (checkResult(build)) {
+                        completed << build
+                    } else {
+                        println "Waiting for build $build to complete"
+                    }
+                    // Avoid spamming TC by waiting 1s between all builds
+                    sleep(TimeUnit.SECONDS.toMillis(1))
```

Comment:
Why wait here?

Reply:
Because I fear that sending 200 requests to TC within a second may accidentally introduce errors.

Reply:
I don't think it will. A rest API that can't handle a few requests per second would be really terrible.

Reply:
Ok, I removed the sleep. Let's figure it out for real :)